projects/self_feeding/README.md

```diff
@@ -1,6 +1,112 @@
+Here is a minimal command for training on the DIALOGUE task using Human-Human (HH) examples:
+```
+python -u $PARLAIHOME/examples/train_model.py -t self_feeding:dialog:train --model projects.self_feeding.self_feeding_agent:SelfFeedingAgent --model-file $MODEL -bs 128
+```
+
+Or to recreate the results in the paper for training on 131k HH examples with the same hyperparameters that we used, run the following:
+```
+python -u $PARLAIHOME/examples/train_model.py -t self_feeding:dialog:train --model-file $MODEL -tblog true -tbcomment $MODEL -tbmetrics lr,dia_acc,dia_loss,dia_rank -ltim 5 -vtim 10 -vp 10 -m projects.self_feeding.self_feeding_agent:SelfFeedingAgent -cands batch --eval-candidates inline -histsz 2 --embedding-type fasttext_cc --embedding-size 300 --dict-maxtokens 250000 --num-epochs 100 --optimizer adamax --embeddings-scale false -bs 128 --relu-dropout 0 --attention-dropout 0 --n-heads 2 --n-layers 2 -lr 0.0025 --ffn-size 32 --lr-scheduler invsqrt --warmup-updates 500 -vmt dia_acc -vmm max
+```
+
+### Train on DIALOGUE (HH) + DIALOGUE (HB) examples
+To train on both HH and HB DIALOGUE examples, point the model to a train file that includes examples from both sets. For example, if you combined 131k HH DIALOGUE examples and 60k HB dialogue examples into a file called `train_hh131k_hb60k.txt`, you could add the following flag to train on that combined file for the DIALOGUE task:
+```
+--dia-train train_hh131k_hb60k.txt
+```
+
+### Train on DIALOGUE (HH) + FEEDBACK examples
+To train on more than one task (such as DIALOGUE and FEEDBACK), modify the command for training on DIALOGUE (HH) alone as follows:
+- Change `-t self_feeding:dialog:train` to `-t self_feeding:diafee:train`. This will result in a different "teacher" agent being used to train the chatbot, one with access to both 'dia\[logue\]' and 'fee\[dback\]`.
+- Append `,fee_acc,fee_loss` to the end of the `-tbmetrics` argument to include the accuracy and loss terms for the feedback task to the Tensorboard logs.
+- Add `--fee-weight 1.0` to set a coefficient by which the loss for the `feedback` task will be multiplied (the default is 1.0 already).
+
+Putting this all together, the command to recreate the 131k HH + 60k FB result from the paper is as follows (as reported in Table 9 in the paper, this setting had the same optimal hyperparameter settings as 131k HH):
+```
+python -u $PARLAIHOME/examples/train_model.py -t self_feeding:diafee:train --model-file $MODEL -tblog true -tbcomment $MODEL -tbmetrics lr,dia_acc,dia_loss,dia_rank,fee_acc,fee_loss -ltim 5 -vtim 10 -vp 10 -m projects.self_feeding.self_feeding_agent:SelfFeedingAgent -cands batch --eval-candidates inline -histsz 2 --embedding-type fasttext_cc --embedding-size 300 --dict-maxtokens 250000 --num-epochs 100 --optimizer adamax --embeddings-scale false -bs 128 --relu-dropout 0 --attention-dropout 0 --n-heads 2 --n-layers 2 -lr 0.0025 --ffn-size 32 --lr-scheduler invsqrt --warmup-updates 500 -vmt dia_acc -vmm max
```

Comment:
And this failed with:

```python-traceback
[ Saving tensorboard logs here: /tmp/tensorboard ]
Traceback (most recent call last):
  File "examples/train_model.py", line 16, in <module>
    TrainLoop(opt).train()
  File "/private/home/roller/working/parlai/parlai/scripts/train_model.py", line 432, in __init__
    self.writer = TensorboardLogger(opt)
  File "/private/home/roller/working/parlai/parlai/core/logs.py", line 89, in __init__
    self.writer = SummaryWriter(logdir='{}/{}'.format(tbpath, tensorboard_tag))
  File "/private/home/roller/.conda/envs/fairseq-fp16-20190211/lib/python3.6/site-packages/tensorboardX/writer.py", line 279, in __init__
    self.file_writer = FileWriter(logdir=log_dir, **kwargs)
TypeError: type object got multiple values for keyword argument 'logdir'
```

Reply:
dropping tensorboard stuff makes this work so I might just update the readme for now.

Reply:
Likely related to #1820 and a tensorboardX version issue.

Reply:
Dropped the tensorboard arguments from the example commands to temporarily alleviate incompatibility issues.